test/utils/tests-roots.js

```diff
@@ -0,0 +1,18 @@
+const { readdirSync } = require("fs");
```

Comment:
i don't think this file is necessary.
you can have a file for "listSubDirectories" maybe, as a utility.
then inside negative.spec for example, just do:
`it.each(listSubDirectories("./code-smaples/negative"))...`